### ISSUE TITLE
Allow custom edges with undefined handles 

### ISSUE THREAD

**Issue by leny-mi** [wbkd/react-flow#2904]
### Please describe the feature that you want to propose

Hi! I am currently using nodes that can have arbitrary connections in arbitrary directions. For this, I am using a set of handles at convenient positions, and a custom edge type that connects to the _closest_ handle on the target instead of any predefined handle. This allows uesrs to reorder nodes with the connections switching the the appropriate handles.

Now since the graphs I'd like to represent are previously defined in an abstract format, and then loaded into react flow, I do not know what handles the nodes were using when connected. I tried keeping the handles undefined, but this returns the following error.

> [React Flow]: Couldn't create edge for source handle id: "world", edge id: et1-p2. Help: https://reactflow.dev/error#008

I have found the following segment of code that causes this issue (core/dist/esm/index.ts).

```ts
const sourceHandle = getHandle(sourceHandleBounds.source, edge.sourceHandle);
const targetHandle = getHandle(targetNodeHandles, edge.targetHandle);
const sourcePosition = sourceHandle?.position || Position.Bottom;
const targetPosition = targetHandle?.position || Position.Top;
const isFocusable = !!(edge.focusable || (edgesFocusable && typeof edge.focusable === 'undefined'));
if (!sourceHandle || !targetHandle) {
    onError?.('008', errorMessages['008'](sourceHandle, edge));
    return null;
}
```

In my case, simply removing the check is enough as the edges will automatically connect to whatever handles they deem suitable. Would it be possible to add a flag or a parameter to skip this check? Or move it into the base edge class, such that any other edge type can handle it on their own?

**Comment by moklick** [comment]
Thanks for your feedback. Can you check with v11.6.0 ?

**Comment by leny-mi** [comment]
Unless I missed anything added with v11.6.0 that I could use, the issue persists :pensive:

**Comment by moklick** [comment]
When you specify a sourceHandle or targetHandle you need to make sure that you have a handle with that id. In your example you would need a handle of type "source" with the id "world". Is this what you are trying to achieve https://reactflow.dev/docs/examples/edges/simple-floating-edges/ ?

**Comment by leny-mi** [comment]
> Is this what you are trying to achieve https://reactflow.dev/docs/examples/edges/simple-floating-edges/ ?
I have a similar setup, where in my case I connect to the closest handle by euclidean distance. In my case I also have a very large set of types of nodes, where each has a respective custom node component. 

Currently, I need to ensure that every single custom node has a handle with a specific id, in my case "default". While this workaround is fine, I would like users, that will add their own custom node types, not to have to consider such an "arbitrary" constraint. 

An alternative workaround I was thinking about was adding an invisible and unusable handle to every custom node just to ensure that every node has a "default" handle.

So to summarize, in my case I do not know what handle ids the connected nodes will have at the time of creating the edge, but since the edge does not care about the handle ids, it would be ideal to somehow turn off the check I mentioned initally.

**Comment by moklick** [comment]
why can't you leave the sourceHandle and targetHandle empty? It should work if they are not defined at all.

**Comment by leny-mi** [comment]
Because the following piece of code disallows empty sourceHandles and targetHandles even if they are not needed in the custom edge component. 

https://github.com/wbkd/react-flow/blob/29790476d800269e363c8a5cae69cb3cbb39b3bf/packages/core/src/container/EdgeRenderer/index.tsx#L112-L122

Applying a patch that removes lines 118-122 fixes the issue, but I assume that you wouldn't want to entirely remove the error handling.

**Comment by moklick** [comment]
Even if `edge.sourceHandle` is undefined, `sourceHandle` can be defined. The `getHandle` function returns the first handle inside `sourceHandleBounds.source` when `edge.sourceHandle` is not set.
